weaviate/collections/aggregations/hybrid/executor.py

```diff
@@ -10,12 +9,14 @@
 )
 from weaviate.collections.classes.filters import _Filters
 from weaviate.collections.filters import _FilterToGRPC
+from weaviate.connect import executor
+from weaviate.connect.v4 import ConnectionType
 from weaviate.exceptions import WeaviateUnsupportedFeatureError
 from weaviate.types import NUMBER
 
 
-class _HybridAsync(_AggregateAsync):
-    async def hybrid(
+class _HybridExecutor(Generic[ConnectionType], _BaseExecutor[ConnectionType]):
```

Comment:
Do we need the Generic[ConnectionType] here? That should already be present through `_BaseExecutor[ConnectionType`?

Reply:
Yes, it's required because of Python's `<3.12` generics implementation. Once that becomes the minimum then this will instead look like:
```python
class _HybridAsync[C extends ConnectionSync | ConnectionAsync](_BaseExecutor[C]):
    ...
```
which is a much more modern/typical way of defining generics

Reply:
Either way, if the parent class is generic over some type `C` then the child that inherits from it must also be generic over `C` otherwise the child isn't generic. It becomes a specific instance of the generic with some undefined type `C`